libcodechecker/server/session_manager.py

```diff
@@ -393,77 +361,43 @@ def create_or_get_session(self, auth_string):
-
-            # If the session is still valid and credentials are resent,
-            # return old token. This is fetched either locally or from the db.
-            session_or_token = self._fetch_session_or_token(sess_hash)
-            if session_or_token:
-                if isinstance(session_or_token, _Session):
-                    # We were able to fetch a session from the local in-memory
-                    # storage. Use this session, and mark that it has been
-                    # used again, updating the access timestamp.
-                    local_session = session_or_token
-                    local_session.revalidate()
-                elif isinstance(session_or_token, basestring):
-                    # The database gave us a token, which we will reuse in
-                    # creating a local cache entry for the session.
-                    db_token = session_or_token
-
-            # A Session object does not exist in the server memory, so it must
-            # be created now.
-            if not local_session:
-                # Use the previously existing token if one was found in the
-                # database for the user, otherwise generate a brand new one.
-                token = db_token if db_token else \
-                    uuid.UUID(bytes=os.urandom(16)).__str__().replace('-', '')
-
-                user_name = validation['username']
-                groups = validation.get('groups', [])
-                is_root = validation.get('root', False)
-
-                local_session = _Session(
-                    token, sess_hash, user_name, groups,
-                    self.__auth_config['session_lifetime'],
-                    is_root, self.__database_connection)
-                self.__sessions.append(local_session)
-
-                if self.__database_connection:
-                    try:
-                        transaction = self.__database_connection()
-
-                        if not db_token:
-                            # The session created locally was done with a brand
-                            # new token, so the cookie must be saved to the
-                            # database.
-                            record = SessionRecord(sess_hash, token)
-                            transaction.add(record)
-                        else:
-                            # The local session was created with a token that
-                            # was found in the database. In this case, a new
-                            # row cannot be added, but we must update the
-                            # timestamp of the user's access.
-                            transaction.query(SessionRecord) \
-                                .filter(SessionRecord.token == db_token) \
-                                .update(
-                                    {SessionRecord.last_access:
-                                     datetime.now()})
+            if not validation:
```

Comment:
I think some renaming would be better here. One of the validation is root validation and the other is not.

Reply:
It's not one and the other, technically there should be only *one* `validation` object at this line commented. Root authentication should always take priority when a user is logging in.